docs/feature/storage/indexing-and-storage.md

```diff
@@ -0,0 +1,304 @@
+  nodes from the left subtree. Finally, the query output is: {7,11} and {8,9}.
+
+## Doc Values
+
+Until Lucene 4.0 columns were indexed using an inverted index data structure
+that maps terms to document ids. For searching documents by terms, this approach
+is effective and well-suited.
+However, if we have to find field values given document id, this solution
+was not equally efficient.
+Furthermore, to perform column-oriented
+retrieval of data, it was necessary to traverse and extract all fields that
+appear in the collection of documents. This can cause memory and performance
+issues if we need to extract a large amount of data.
+
+To improve the performance of aggregations and sorting, a new data structure was
+introduced, namely Doc Values. Doc Values is a column-based data storage built
+at document index time. They store all field values that are not analyzed as
+strings in a compact column making it more effective for sorting and
+aggregations.
+
+CrateDB implements Column Store based on Doc Values in Lucene. The Column Store
+is created for each field in a document and generated as the following
+structures for fields in the Product table:
+
+|           | **Document 1** | **Document 2** | **Document 3** |
+| --------- | -------------- | -------------- | -------------- |
+| productID | 1              | 2              | 3              |
+| name      | Almond Milk    | Almond Flour   | Milk           |
+| quantity  | 100            | 200            | 300            |
+
+For example, for the first document, CrateDB creates the following mappings as
+Column Store: {productID → 1, name → “Almond Milk“, quantity → 100}.
+
+Column Store significantly improves aggregations and grouping as the data for
+one column is packed in one place. Instead of traversing each document and
+fetching values of the field that can also be very scattered, we extract all
+field data from the existing Column Store. This approach significantly improves
+the performance of sorting, grouping, and aggregation operations. In CrateDB,
+Column Store is enabled by default and can be disabled only for text fields, not
+for other primitive types. Furthermore, CrateDB does not support storing values
+for {ref}`container <container>` and {ref}`geographic <geospatial>` data types
+in Column Store.
+
+Besides fields, CrateDB also supports Column Store for the JSON representation
+of each row in a table. For our example, row-based Column Store is generated as
+the following:
+
+| **Document** | **Row**                                         |
+| ------------ | ----------------------------------------------- |
+| 1            | {“id“:1, “name“:”Almond Milk”, “quantity“:100}  |
+| 2            | {“id“:2, “name“:”Almond Flour”, “quantity“:200} |
+| 3            | {“id“:3, “name“:”Milk”, “quantity“:300}         |
+
+The use of Column Store results in a small disk footprint, thanks to specialized
+compression algorithms such as delta encoding, bit packing, and GCD.
```

Comment:
I think it's a good idea to educate how CrateDB's column store builds upon Lucene's doc values. However, the section is long and could be refurbished, possibly splitting it into two or three blocks/rubrics.